ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -524,24 +529,105 @@ KEXT_STATIC int HandleFileOpOperation(
                 goto CleanupAndReturn;
             }
         }
-        else
+    }
+    else if (KAUTH_FILEOP_LINK == action)
+    {
+        const char* fromPath = reinterpret_cast<const char*>(arg0);
+        const char* newPath = reinterpret_cast<const char*>(arg1);
+        
+        // TODO(Mac): We need to handle failures to lookup the vnode.  If we fail to lookup the vnode
+        // it's possible that we'll miss notifications
+        errno_t toErr = vnode_lookup(newPath, 0 /* flags */, &currentVnode, context);
+        if (0 != toErr)
         {
-            PerfSample hardLinkSample(&perfTracer, PrjFSPerfCounter_FileOp_HardLinkCreated);
+            KextLog_Error("HandleFileOpOperation: vnode_lookup failed, errno %d for path '%s'", toErr, newPath);
```

Comment:
Now that this can be logged in two spots (here and line 476) it would be helpful to include the `action` that the kext was processing in the error message.